### DISCUSSION TITLE
Include Kakao OIDC provider

### DISCUSSION THREAD

**Discussion by Casamatechs** [nextauthjs/next-auth#7283] [Ideas]
### Description 📓

Right now the Kakao provider only supports OAuth2.0 for authentication, but Kakao also allows developers to activate OpenID connect.

![image](https://user-images.githubusercontent.com/15941476/232279922-a503a181-4957-4cf1-8822-038b61323983.png)

Since the data returned to the callback is different depending on the kind of authentication method and by default Kakao apps don't allow OpenID protocol, I think that is better to have a totally separated provider (`kakao-oidc`) rather than adding logic to the existing Kakao provider, which could lead to a more difficult to maintain codebase.

I have a working PoC and I am willing to also update the docs for next-auth v4 if this proposal is accepted.

### How to reproduce ☕️

N/A

### Contributing 🙌🏽

Yes, I am willing to help implement this feature in a PR

**Comment by sean-ahn** [comment]
I have created a proposal for this: https://github.com/nextauthjs/next-auth/discussions/8107.
